build/tools/Automation/Python/Python.cs

```diff
@@ -0,0 +1,159 @@
+                    !oryxSdkVersions.Contains(newVersion))
+                {
+                    Console.WriteLine($"newVersion {newVersion}");
+                    versionObjs.Add(new VersionObj
+                    {
+                        Version = newVersion,
+                        GpgKey = this.GetGpgKeyForVersion(newVersion),
+                    });
+                }
+            }
+
+            return versionObjs;
+        }
+
+        private void UpdateOryxConstantsForNewVersions(List<VersionObj> versionObjs, List<YamlConstants> yamlConstants, string oryxRootPath)
+        {
+            Dictionary<string, YamlConstants> pythonYamlConstants = this.GetYamlPythonConstants(yamlConstants);
+
+            foreach (var versionObj in versionObjs)
+            {
+                string version = versionObj.Version;
+                string pythonConstantKey = this.GeneratePythonConstantKey(versionObj);
+                Console.WriteLine($"[UpdateConstants] version: {version} pythonConstantKey: {pythonConstantKey}");
+                pythonYamlConstants["python-versions"].Constants[pythonConstantKey] = version;
+
+                this.UpdateVersionsToBuildTxt(versionObj, oryxRootPath);
+            }
+
+            var serializer = new SerializerBuilder()
+                .WithNamingConvention(UnderscoredNamingConvention.Instance)
+                .Build();
+
+            var constantsYamlAbsolutePath = Path.Combine(oryxRootPath, "build", Constants.ConstantsYaml);
+            var stringResult = serializer.Serialize(yamlConstants);
+            File.WriteAllText(constantsYamlAbsolutePath, stringResult);
+        }
+
+        private string GetGpgKeyForVersion(string version)
+        {
+            // Split the version string into major and minor parts
+            string[] parts = version.Split('.');
+            if (parts.Length < 2)
+            {
+                throw new ArgumentException("Invalid version format");
+            }
+
+            string majorMinor = string.Join(".", parts.Take(2));
+
+            // Look up the GPG key in the dictionary
+            if (Constants.VersionGpgKeys.TryGetValue(majorMinor, out string gpgKey))
+            {
+                return gpgKey;
+            }
+            else
+            {
+                throw new ArgumentException($"GPG key not found for version {version}.");
+            }
+        }
+
+        private string GeneratePythonConstantKey(VersionObj versionObj)
+        {
+            string[] splitVersion = versionObj.Version.Split('.');
+            string majorVersion = splitVersion[0];
+            string minorVersion = splitVersion[1];
+
+            // prevent duplicate majorMinor where 11.0 and 1.10 both will generate a 110 key.
+            int majorVersionInt = int.Parse(majorVersion);
+            string majorMinor = majorVersionInt < 10 ? $"{majorVersion}{minorVersion}" : $"{majorVersion}Dot{minorVersion}";
```

Comment:
Does the key need to follow a specific format here? Any reason we can't just do `$"{majorVersion}_{minorVersion}"` ?

Reply:
Updated to use `_`, since no specific format is needed.